pkg/fsutils/basepath.go

```diff
@@ -0,0 +1,77 @@
+package fsutils
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/ldez/grignotin/goenv"
+)
+
+// Relative path modes.
+const (
+	RelativePathModeGoMod   = "gomod"
+	RelativePathModeGitRoot = "gitroot"
+	RelativePathModeCfg     = "cfg"
+	RelativePathModeWd      = "wd"
+)
+
+func AllRelativePathModes() []string {
+	return []string{RelativePathModeGoMod, RelativePathModeGitRoot, RelativePathModeCfg, RelativePathModeWd}
+}
+
+func GetBasePath(ctx context.Context, mode, cfgDir string) (string, error) {
+	if mode == "" {
+		// TODO(ldez): v2 the default should be cfg or gomod.
+		mode = RelativePathModeWd
+	}
+
+	switch mode {
+	case RelativePathModeCfg:
+		if cfgDir == "" {
+			return GetBasePath(ctx, RelativePathModeWd, cfgDir)
+		}
+
+		return cfgDir, nil
+
+	case RelativePathModeGoMod:
+		goMod, err := goenv.GetOne(ctx, goenv.GOMOD)
+		if err != nil {
+			return "", fmt.Errorf("get go.mod path: %w", err)
+		}
+
+		return filepath.Dir(goMod), nil
+
+	case RelativePathModeGitRoot:
+		root, err := gitRoot(ctx)
+		if err != nil {
+			return "", fmt.Errorf("get git root: %w", err)
+		}
+
+		return root, nil
+
+	case RelativePathModeWd:
+		wd, err := Getwd()
+		if err != nil {
+			return "", fmt.Errorf("get wd: %w", err)
+		}
+
+		return wd, nil
+
+	default:
+		return "", errors.New("unknown relative path mode")
+	}
+}
+
+func gitRoot(ctx context.Context) (string, error) {
+	cmd := exec.CommandContext(ctx, "git", "rev-parse", "--show-toplevel")
```

Comment:
This is a new dependency right, is it reasonable to add `git` as a dependency? Probably worth mentio this in the future close to where the mode is described.

Reply:
we already have `git` as "dependency": the diff processor is based on it.

Reply:
Also, this is only called if the mode `gitroot` is used.

The modes are documented.

Reply:
Ah, I tried to look for any other places but couldn't find one, at least not one that shells out like this.

Reply:
This is inside `revgrep`: https://github.com/golangci/revgrep/blob/main/patch.go